src/index.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+var crypto = require('crypto');
+var update = require('./update');
+var log = require('./logger');
+
+module.exports = function (keyword, limit) {
+  if (typeof keyword !== 'string') {
+    log.error('Index: Keyword is required');
+    return false;
+  }
+
+  var etag = null;
+  var cache = {};
+
+  this.getETag = () => etag;
+  this.getCache = () => cache;
+
+  this.update = () => new Promise((resolve, reject ) => {
```

Comment:
mostly because npmKeywordSearch is what is creating the internal promise and I didn't want to rely on that being a promise. I suppose its safe to do that though